src/execution_engines/node_js/execute.ts

```diff
@@ -10,40 +10,57 @@ async function getBrowser () {
   return browser
 }
 
-export async function loadPage () {
+export async function loadPage (executable: string) {
   const browser = await getBrowser()
-  return browser.newPage()
+  const page = await browser.newPage()
+
+  // Disallow all outgoing requests
+  await page.setRequestInterception(true)
+  page.on('request', interceptedRequest => {
+    interceptedRequest.abort()
+  })
+
+  await page.evaluate(({ executable }) => {
+    /* eslint-disable */
+    const exec = Function(`"use strict"; return (${executable})`)()
+    /* eslint-enable */
+    const w: any = window
+    w.contract = exec
+  }, { executable })
+  return page
 }
 
 export function unloadPage (page: puppeteer.Page) {
```

Comment:
I think this might be the right spot for it is because if we consider this unload behaviour, it is purely a wrapper around `page.close`. The catch block says, "if the page is already closed, don't worry about it that's fine". So it feels like the right place imo.

Reply:
It's true we could remove this function completely, and call `.close` from the upper context as you say. I kind of like keeping the catch out of the upper engine

Reply:
👍